hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -562,6 +563,14 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
+	// Default the infraID if unset
+	if hcluster.Spec.InfraID == "" {
+		hcluster.Spec.InfraID = fmt.Sprintf("%s-%s", hcluster.Name, utilrand.String(5))
```

Comment:
we should reuse the installer's code for generating it here and in the CLI
https://github.com/openshift/installer/blob/3d19350885d593ee2b1d9ecd7612c2d697dab2a3/pkg/asset/installconfig/clusterid.go#L46-L79

Reply:
should we also "fail" if is empty for aws?

Reply:
Not sure, would anything break if we generated here it in the AWS case?

Updated to re-use the installer code.

Reply:
We'd have to try. It wouldn't match any tags on infrastructure we've pre-created. I think we can leave as is and if necessary add the additional check for AWS later.

Reply:
>It wouldn't match any tags on infrastructure we've pre-created

right, I don't think we want to support that scenario neither we are testing it so my vote would be to fail in a follow up.